app/cmd.go

```diff
@@ -657,6 +659,17 @@ func getConfig(ctx context.Context) (Config, error) {
 		UIDir: viper.GetString("ui-dir"),
 	}
 
+	if cfg.PublicURL != "" {
+		u, err := url.Parse(cfg.PublicURL)
+		if err != nil {
+			return cfg, errors.Wrap(err, "parse public url")
+		}
+		if cfg.HTTPPrefix != "" {
+			return cfg, errors.New("public-url and http-prefix cannot be used together")
```

Comment:
is this because we are using the HTTP Prefix as an env variable when we build the public URL, if set?

Reply:
The path of the `--public-url` flag is used _AS_ the prefix, and so if you specified both it would be ambiguous which one should take precedence so we have it bail out if we detect both.

Since `--public-url` is a new flag, this shouldn't cause existing setups to fail, and anyone migrating to the new flag will get an error if they forget to unset the old one (if they are using it). Safe upgrade path, with a fail-to-start scenario if misconfigured.